ios/browser/BUILD.gn

```diff
@@ -15,37 +15,29 @@ source_set("browser") {
   ]
 
   deps = [
-    "api/bookmarks",
-    "api/bookmarks/exporter",
-    "api/bookmarks/importer",
-    "api/brave_shields",
-    "api/brave_wallet:wallet_mojom_wrappers",
-    "api/certificate",
-    "api/favicon",
-    "api/history",
-    "api/ipfs",
-    "api/net",
-    "api/ntp_background_images",
-    "api/opentabs",
-    "api/p3a",
-    "api/password",
-    "api/session_restore",
-    "api/sync",
-    "api/url",
-    "api/version_info",
-    "api/web",
-    "api/web_image",
-    "brave_wallet",
+    ":chrome_ios",
     "//base",
     "//brave/browser/sync",
     "//brave/chromium_src/ios/chrome/browser/shared/model/prefs",
+    "//brave/components/constants",
+    "//brave/ios/browser/api",
     "//brave/ios/browser/application_context",
+    "//brave/ios/browser/brave_wallet",
     "//brave/ios/browser/browser_state",
+    "//brave/ios/browser/ui/webui",
     "//components/flags_ui",
     "//components/metrics",
     "//components/metrics_services_manager",
     "//components/variations",
     "//components/variations/service",
+    "//ui/base",
+    "//url",
+  ]
+}
+
+component("chrome_ios") {
+  visibility = [ ":browser" ]
```

Comment:
`visibility` restriction should be added into `//brave/components/webui` target.